### PR TITLE
AnonFunc, Notrly and DependencyUpdates

### PR DESCRIPTION
**Pull request by AnEmortalKid**
Fixed todos in readme from test dir.

Added support for:
```
notrly blah
  shh 1
wow
``` 

producing:
```
if (! blah) {
  //1
}
```

- Cleaned up test package
- Added debugging harness that produces a page with results for easier debugging through chrome
- Added exerciser (like a debugger but you type code)
- Added creation of anonymous function `such much a b` -> `function(a, b){}`

### CODE REVIEW

**Comment by AnEmortalKid** [comment]
@alistairyves you might want some of these :) i think im slowly figuring out what to do

**Comment by AnEmortalKid** [comment]
@Bartvds mind merging this for us?

**Comment by Bartvds** [comment]
I dunno, @remixz maybe?

File: example/WOWserify/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# WOWserify
+
+A sample page displaying dogescript on one text area, and javascript in another.
```
Comment (Pholey) [inline example/WOWserify/README.md]: I opened a pull request to this branch here https://github.com/AnEmortalKid/dogescript/pull/3; I'd love to remove all of this generated code from git if possible

**Comment by AnEmortalKid** [comment]
Done! Thanks a lot!

**Comment by Pholey** [comment]
Took a more in depth look and added a couple test cases here https://github.com/AnEmortalKid/dogescript/pull/4 

Aside from that it looks good to me and I'm happy to merge this. Great job 😄 I would like to give @remixz another ping and a bit more time to give it a once over before doing so however 👍 

As a side note, i think in the future we could reuse the wowserify code for the deboogering process rather than having two very similar copies for debugging/example code, for now though having something is most certainly better than nothing at the moment i think.

**Comment by AnEmortalKid** [comment]
Yeah, I  did notice there was some crossover, I didn't know what I was
doing at first so I based things off of the wowserify bit while I learned
what the hell was going on with NPM and scripts and all that. I'd probably
like to merge the exerciser/debugger thing, since it feels like they both
kinda do the same thing. The debugging page really would only have to have
the text area with dogescript and the one with javascript and a button to
force parsing (after someone puts in a breakpoint).

I'll see if I can clean it up :)

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by Pholey** [comment]
After some thinking i'd really like to move all of the wowserfy/deboogering code to it's [own separate repository](https://github.com/dogescript/deflea) and link to it in the documentation instead (possibly include it as a git-submodule if we really want it in the repo for development processes) I just don't think it makes all too much sense being a part of the parser code when it's simply an `npm link` away from another folder. If you are comfortable enough with react and would be open to porting over your debugger code @AnEmortalKid i'm happy to merge it in over there as soon as you are done. Otherwise i can port it over when i get some time later today or tomorrow.

@dogescript/owners Any thoughts about this?

**Comment by AnEmortalKid** [comment]
I haven't done react at all, I could try to give it a go. Really, the only thing we'd probably need to port over is:

```javascript
rundoge = function(src)
{
  window.eval(dogescript(src));
 } 
```

and the button:
` <button onclick="rundoge(document.getElementsByClassName('dogescript')[0].value)">Run code()</button>`


The other two scrips were basically just loading the dogescript / expected javascript and updating the text content. 

Maybe we could have a 'import file' option that would let you pick a dogescript source?

**Comment by alistairyves** [comment]
I dont think its necessary to put tests in another repo, I believe its possible to remove the test files at the time of publishing via the npm publish script

**Comment by AnEmortalKid** [comment]
I think this would be just to remove the debugger and exerciser and
wowserify code, which are supplemental to debugging the tests and parser.
The tests would stay here.

**Comment by Pholey** [comment]
@alistairyves Agreed 100%, don't want to move the testing code out of dogescript at all, just the "example" page that is currently not really related to the parser aside from how to actually use dogescript

**Comment by AnEmortalKid** [comment]
no worries, I was in the same boat :)

**Comment by Pholey** [comment]
Also @AnEmortalKid can we add `such much` to the language spec as well here?

**Comment by AnEmortalKid** [comment]
added: https://github.com/dogescript/dogescript/pull/104/commits/0b9d5f2fc1bf9077eb51335ca9784ea3979b650d

**Comment by Pholey** [comment]
Awesome thanks @AnEmortalKid. Got the [test importer  working with deflea](https://github.com/dogescript/deflea/blob/master/scripts/import-specs.js) last night, now all that's left is actually styling it and improving the dev process a bit, and removing the debooger code from this PR (I'll open a PR after i'm all done with deflea if you don't feel like doing it)

**Comment by alistairyves** [comment]
ok @Pholey I understand what you mean, it looks good, I've been quite busy, but I am going to take a look this afternoon

@AnEmortalKid sorry i've been unresponsive so long

**Comment by AnEmortalKid** [comment]
No worries, I know we all unfortunately have a life outside of dogescript :)

**Comment by AnEmortalKid** [comment]
@alistairyves we started a slack (dogescript.slack.com). This app should load in a bit and be able to send you an invite: https://doge-invite.herokuapp.com/

File: test/README.md [hunk 2]
```diff
@@ -13,10 +13,20 @@ node ./test/spec.test.js | tap-unfunk
...
+1. Optionally, create a `scenario.desc` file describing the test, this will be displayed if the test fails.
 
 To keep the tests sane use foo/bar/baz/buzz as custom identifiers and strings (as they are not doge-like).
 
-## Todo
+# Debug package
 
-- fix notrly / but notrly
-- make separate case for operators used in rly/much/many
+There are new npm scripts that help with debugging code:
```
Comment (Pholey) [inline test/README.md]: Can we remove this? We have this over in `deflea` now :)

File: test/debug/debugger-clean.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+var fs = require('fs');
```
Comment (Pholey) [inline test/debug/debugger-clean.js]: We can kill this file too

File: test/debug/debugger-page.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<html>
```
Comment (Pholey) [inline test/debug/debugger-page.html]: 🔥

File: test/debug/debugger-setup.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+dogescript = require('../../index');
```
Comment (Pholey) [inline test/debug/debugger-setup.js]: 🔥

File: test/debug/exerciser-clean.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+var fs = require('fs');
```
Comment (Pholey) [inline test/debug/exerciser-clean.js]: 🔥

File: test/debug/exerciser.html [hunk 7]
```diff
@@ -0,0 +1,44 @@
+<html>
```
Comment (Pholey) [inline test/debug/exerciser.html]: 🔥

File: test/debug/exerciser.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+dogescript = require('../../index');
+
```
Comment (Pholey) [inline test/debug/exerciser.js]: 🔥

File: test/debugmaker.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
+var fs = require('fs');
+var path = require('path');
```
Comment (Pholey) [inline test/debugmaker.js]: 🔥

**Comment by AnEmortalKid** [comment]
:fire: up!

**Comment by Pholey** [comment]
made a small change to your PR https://github.com/dogescript/dogescript/pull/104/commits/9218314d53f303d6980997b2cd9dc23297a4b7e0 , Would you mind reviewing it? After that i think we are ready to hit that green merge button 👍

**Pull request merged**